### PR TITLE
Add new OSS::PhoneNumberInput component

### PR DESCRIPTION
**Pull request by Miexil**
### What does this PR do?

Adds a new PhoneNumberInput component.

Related to: [#1269](https://github.com/upfluence/backlog/issues/1269)

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: app/styles/phone-number-input.less [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  .phone-number-input {
+    .country-selector {
+      border-right: 1px solid @upf-gray;
+      height: 100%;
+      display: flex;
+      align-items: center;
+      padding-right: 12px;
+      margin-right: 12px;
```
Comment (JulienVannier66) [inline app/styles/phone-number-input.less:26]:
```suggestion
      padding-right: @spacing-xx-sm;
      margin-right: @spacing-xx-sm;
```

File: addon/components/o-s-s/phone-number-input.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { countries, CountryData } from '@upfluence/oss-components/utils/country-codes';
+
+interface OSSPhoneNumberInputArgs {
+  prefix: string;
+  number: string;
+  onChange: (prefix: string, number: string) => void;
```
Comment (JulienVannier66) [inline addon/components/o-s-s/phone-number-input.ts:9]:
```suggestion
  onChange(prefix: string, number: string): void;
```
We don't use this syntax now?

File: app/styles/phone-number-input.less (hunk 1 above)
Comment (Miexil) [reply app/styles/phone-number-input.less:26]: That's not the same width sadly, I tried using our variables but couldn't get something that matched exactly.

File: addon/components/o-s-s/phone-number-input.ts (hunk 2 above)
Comment (Miexil) [reply addon/components/o-s-s/phone-number-input.ts:9]: We sure do !

File: app/styles/phone-number-input.less (hunk 1 above)
Comment (JulienVannier66) [reply app/styles/phone-number-input.less:26]: Ok ok no problem, it's good for me so

**Review by JulienVannier66 — APPROVED** [review]

File: addon/components/o-s-s/phone-number-input.hbs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<div class="phone-number-container fx-1" ...attributes>
+  <div class="phone-number-input upf-input fx-row fx-1 fx-xalign-center">
+    <div class="country-selector fx-row" role="button" {{on "click" this.toggleCountrySelector}}>
+      <div class="fflag fflag-{{this.selectedCountry.id}} ff-sm ff-wave ff-rounded"></div>
+      {{#if this.countrySelectorShown}}
+        <i class="fal fa-chevron-up margin-left-xxx-sm"></i>
```
Comment (phndiaye) [inline addon/components/o-s-s/phone-number-input.hbs:6]: so we start using the `fal` ?

Comment (Miexil) [reply addon/components/o-s-s/phone-number-input.hbs:6]: In this case it is just a thinner chevron ; is that an issue ? 🤔

**Review by phndiaye — APPROVED** [review]

**Pull request merged**